src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -343,27 +337,23 @@ def check_helm_install(kube_config, kube_context):
         _, error_helm_installed = response_helm_installed.communicate()
         if response_helm_installed.returncode != 0:
             if "unknown flag" in error_helm_installed.decode("ascii"):
-                telemetry.set_user_fault()
                 telemetry.set_exception(exception='Helm 3 not found', fault_type=consts.Helm_Version_Fault_Type,
                                         summary='Helm3 not found on the machine')
-                raise CLIError("Please install the latest version of Helm. " +
-                               "Learn more at https://aka.ms/arc/k8s/onboarding-helm-install")
-            telemetry.set_user_fault()
+                raise ClientRequestError("Helm 3 not found", recommendation="Please install the latest version of Helm. " +
+                                         "Learn more at https://aka.ms/arc/k8s/onboarding-helm-install")
             telemetry.set_exception(exception=error_helm_installed.decode("ascii"), fault_type=consts.Helm_Installation_Fault_Type,
                                     summary='Helm3 not installed on the machine')
-            raise CLIError(error_helm_installed.decode("ascii"))
+            raise DeploymentError(error_helm_installed.decode("ascii"))
```

Comment:
Use ValidationError here. DeploymentError is when ARM template deployment fails. https://github.com/Azure/azure-cli/blob/119856d4f5984e5a4ef0cdb5bc233c6f07da2720/src/azure-cli-core/azure/cli/core/azclierror.py#L230